src/main/java/gregtech/common/pipelike/fluidpipe/tile/TileEntityFluidPipeTickable.java

```diff
@@ -198,17 +199,28 @@ public void checkAndDestroy(@Nonnull FluidStack stack) {
 
         boolean burning = prop.getMaxFluidTemperature() < fluid.getTemperature(stack);
         boolean leaking = !prop.isGasProof() && fluid.isGaseous(stack);
-        boolean shattering = !prop.isCryoProof() && fluid.getTemperature() < IPropertyFluidFilter.CRYOGENIC_TEMPERATURE_THRESHOLD;
+        boolean shattering = !prop.isCryoProof() && fluid.getTemperature() < FluidConstants.CRYOGENIC_FLUID_THRESHOLD;
         boolean corroding = false;
         boolean melting = false;
 
-        if (fluid instanceof MaterialFluid materialFluid) {
-            corroding = !prop.isAcidProof() && materialFluid.getFluidType().equals(FluidTypes.ACID);
-            melting = !prop.isPlasmaProof() && materialFluid.getFluidType().equals(FluidTypes.PLASMA);
+        if (fluid instanceof AttributedFluid attributedFluid) {
+            FluidState state = attributedFluid.getState();
+            if (!prop.canContain(state)) {
+                leaking = state == FluidState.GAS;
+                melting = state == FluidState.PLASMA;
+            }
 
             // carrying plasmas which are too hot when plasma proof does not burn pipes
-            if (burning && prop.isPlasmaProof() && materialFluid.getFluidType().equals(FluidTypes.PLASMA))
+            if (burning && prop.canContain(FluidState.PLASMA)) {
                 burning = false;
+            }
```

Comment:
This could just be moved to the declaration of `burning`.

```java
burning = prop.getMaxFluidTemperature() < fluid.getTemperature(stack) && !prop.canContain(FluidState.PLASMA)
```